modules/data.land/R/ic_process.R

```diff
@@ -119,9 +138,41 @@ ic_process <- function(settings, input, dir, overwrite = FALSE){
     if(input$output == "css"){
       settings$run$inputs[["pss"]][['path']]  <- gsub("css","pss", path_to_settings)
       settings$run$inputs[["site"]][['path']] <- gsub("css","site", path_to_settings)
+      
+      # IF: For now IC workflow is only working for ED and it's the only case for copying to remote
+      # but this copy to remote might need to go out of this if-block and change
+      
+      # Copy to remote and change paths if needed
+      if (settings$host$name != "localhost") {
+        
+        out.dir.remote   <- file.path(settings$host$folder, paste0(new.site$name, "_", input$source))
+        pss.file.remote  <- file.path(out.dir.remote, basename(settings$run$inputs[["pss"]][['path']]))
+        settings$run$inputs[["pss"]][['path']] <- pss.file.remote
+
+        css.file.remote  <- file.path(out.dir.remote, basename(settings$run$inputs[["css"]][['path']]))
+        settings$run$inputs[["css"]][['path']] <- css.file.remote
+        
+        site.file.remote <- file.path(out.dir.remote, basename(settings$run$inputs[["site"]][['path']]))
+        settings$run$inputs[["site"]][['path']] <- site.file.remote
+        
+        remote.execute.cmd(settings$host, "mkdir", c("-p", out.dir.remote))
+        PEcAn.utils::remote.copy.to(settings$host, pss.file.local, pss.file.remote)
+        PEcAn.utils::remote.copy.to(settings$host, css.file.local, css.file.remote)
+        PEcAn.utils::remote.copy.to(settings$host, site.file.local, site.file.remote)
+        
+        # update DB record
+        ic_input <- db.query(paste0("SELECT * from inputs where id = ", putveg.id), con)
+        fname    <- db.query(paste0("SELECT * from formats where id = ", ic_input$format_id), con)
+        mtype    <- db.query(paste0("SELECT type_string from mimetypes where id = ", fname$mimetype_id), con)
+        dbfile.input.insert(in.path = out.dir.remote, in.prefix = model_file[["file_name"]],
```

Comment:
...plus a line that queries the input to get it's ID, plus a line or two that pulls out the paths on the local and remote site, plus checking for errors and edge cases. Doesn't have to be a huge function, but I think it would be convenient to have a function that just takes in the input.id, remote folder, and con and handles both file transfer and DB updating. Might return the remote dbfile record so you have the IDs and remote path and prefix.